packages/calcite-components/src/components/shell-panel/shell-panel.tsx

```diff
@@ -242,246 +147,203 @@ export class ShellPanel extends LitElement {
+    const arrowKeys =
+      layout === "horizontal" ? ["ArrowUp", "ArrowDown"] : ["ArrowLeft", "ArrowRight"];
 
-    const horizontalKeys = ["ArrowLeft", "ArrowRight"];
-    const verticalKeys = ["ArrowDown", "ArrowUp"];
-    const directionFactor = dir === "rtl" && horizontalKeys.includes(key) ? -1 : 1;
+    const keys = [...arrowKeys, "Home", "End"];
 
-    const increaseKeys =
-      layout === "horizontal"
-        ? position === "end"
-          ? key === verticalKeys[1] || key === horizontalKeys[0]
-          : key === verticalKeys[0] || key === horizontalKeys[1]
-        : key === verticalKeys[1] ||
-          (position === "end" ? key === horizontalKeys[0] : key === horizontalKeys[1]);
-
-    if (increaseKeys) {
-      const stepValue = event.shiftKey ? multipliedStep : step;
-
-      return layout === "horizontal"
-        ? initialContentHeight + directionFactor * stepValue
-        : initialContentWidth + directionFactor * stepValue;
+    if (!resizable || !contentEl || defaultPrevented || !keys.includes(key)) {
+      return;
     }
 
-    const decreaseKeys =
-      layout === "horizontal"
-        ? position === "end"
-          ? key === verticalKeys[0] || key === horizontalKeys[0]
-          : key === verticalKeys[1] || key === horizontalKeys[1]
-        : key === verticalKeys[0] ||
-          (position === "end" ? key === horizontalKeys[1] : key === horizontalKeys[0]);
-
-    if (decreaseKeys) {
-      const stepValue = event.shiftKey ? multipliedStep : step;
-
-      return layout === "horizontal"
-        ? initialContentHeight - directionFactor * stepValue
-        : initialContentWidth - directionFactor * stepValue;
+    const rect = this.getContentElDOMRect();
+    const invertRTL = getElementDir(el) === "rtl" ? -1 : 1;
+    const stepValue = shiftKey ? resizeShiftStep : resizeStep;
+
+    switch (key) {
+      case "ArrowUp":
+        this.updateSize({
+          size:
+            rect.height + (layout === "horizontal" && position === "end" ? stepValue : -stepValue),
+          type: "blockSize",
+        });
+        event.preventDefault();
+        break;
+      case "ArrowDown":
+        this.updateSize({
+          size:
+            rect.height + (layout === "horizontal" && position === "end" ? -stepValue : stepValue),
+          type: "blockSize",
+        });
+        event.preventDefault();
+        break;
+      case "ArrowLeft":
+        this.updateSize({
+          size:
+            rect.width +
+            (layout === "vertical" && position === "end" ? stepValue : -stepValue) * invertRTL,
+          type: "inlineSize",
+        });
+        event.preventDefault();
+        break;
+      case "ArrowRight":
+        this.updateSize({
+          size:
+            rect.width +
+            (layout === "vertical" && position === "end" ? -stepValue : stepValue) * invertRTL,
+          type: "inlineSize",
+        });
+        event.preventDefault();
+        break;
+      case "Home":
```

Comment:
Could we `preventDefault()` on "Home" and "End" - this is causing page scroll at times in addition to the resize functionality.

Reply:
Yep. that's a mistake for sure.